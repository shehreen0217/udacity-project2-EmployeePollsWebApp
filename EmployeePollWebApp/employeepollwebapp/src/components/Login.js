import React, { useEffect, useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import Avatar from "@mui/material/Avatar";
import loginAvatar from "../avatars/loginavatar.jpg";
import "../index.css";
// import Stack from "@mui/material/Stack";
import TextField from "@mui/material/TextField";
import Autocomplete from "@mui/material/Autocomplete";
import Button from "@mui/material/Button";
import { Redirect } from "react-router-dom";
import { login, settingUsers } from "../Store/Actions/login";
// import Alert from "@mui/material/Alert";
// import Collapse from "@mui/material/Collapse";

const Login = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(settingUsers());
  }, [dispatch]);

  const auth = useSelector((state) => state.auth.isAuthenticated);
  const allusers = useSelector((state) => state.user.allUsers);
  const [user, setUser] = useState("");
  const [password, setPassword] = useState("");
  // const [open, setOpen] = useState(true);

  if (auth) {
    console.log("in redirect if");
    const urlParams = new URLSearchParams(window.location.search);
    const redirectUrl = urlParams.get("redirectTo");
    return <Redirect to={redirectUrl ? redirectUrl : "/"} />;
  }

  console.log("login is gone mad");
  const Cuser = Object.keys(allusers).map((key) => {
    return allusers[key];
  });
  let allUsers = Cuser.map((e) => e.id);

  const clickHandler = () => {
    dispatch(login(user, password));
    // setReload(true);
  };

  return (
    !auth && (
      <div className="Login">
        <span>
          <Avatar
            className="avatar"
            alt="login avatar"
            src={loginAvatar}
            sx={{ width: 150, height: 150 }}
          />
          <h1 className="login-text">Login</h1>
          <form>
            <Autocomplete
              data-testid="username"
              inputValue={user}
              onInputChange={(e, currentUser) => {
                setUser(currentUser);
              }}
              disablePortal
              id="combo-box-demo"
              options={allUsers}
              sx={{ width: 300 }}
              renderInput={(params) => <TextField {...params} label="User" />}
            />
            <TextField
              data-testid="password"
              onChange={(e) => {
                setPassword(e.target.value);
              }}
              name="password"
              type="password"
              placeholder="password"
              label="Password"
              sx={{ width: 300 }}
            />
          </form>

          {user && password && (
            <Button
              alt="submit"
              aria-label="submit"
              data-testid="submit-btn"
              sx={{
                mt: 1, // margin top
              }}
              onClick={clickHandler}
            >
              Log in
            </Button>
          )}
          {!user ||
            (!password && (
              <Button
                alt="submit"
                aria-label="submit"
                data-testid="submit-btn"
                disabled
                sx={{
                  mt: 1, // margin top
                }}
                onClick={clickHandler}
              >
                Log in
              </Button>
            ))}

          {/* {invalidPass && (
            <Collapse in={open}>
              <Alert onClose={() => setOpen(false)} severity="error">
                Invalid Password!
              </Alert>
              <Redirect to="/login" />
            </Collapse>
          )} */}
        </span>
      </div>
    )
  );
};

export default Login;
