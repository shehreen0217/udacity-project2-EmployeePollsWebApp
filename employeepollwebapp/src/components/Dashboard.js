import React from "react";
import { useSelector, useDispatch } from "react-redux";
import Poll from "./poll";
import { question } from "../Store/Actions/question";
import { useEffect } from "react";

const Dashboard = () => {
  const dispatch = useDispatch();
  const initialQuestions = useSelector((state) => state.questions.Questions);
  const currentUser = useSelector((state) => state.user.user);
  const Ques = Object.keys(initialQuestions).map((key) => {
    return initialQuestions[key];
  });

  // const auth = useSelector((state) => state.auth.isAuthenticated);

  useEffect(() => {
    dispatch(question(Ques, currentUser));
  }, [dispatch]);

  const ans = useSelector((state) => state.questions.Answered);
  const unans = useSelector((state) => state.questions.Unanswered);
  const answered = Object.keys(ans).map((key) => {
    return ans[key];
  });
  const unanswered = Object.keys(unans).map((key) => {
    return unans[key];
  });

  answered.sort((x, y) => y.timestamp - x.timestamp);
  unanswered.sort((x, y) => y.timestamp - x.timestamp);

  return (
    <div>
      <h2 style={styles.heading}> New Questions</h2>
      <div style={styles.div}>
        {initialQuestions &&
          unanswered.map((question) => {
            return <Poll key={question.id} question={question} value="Poll" />;
          })}
      </div>
      <br></br>
      <h2 style={styles.heading}> Answered</h2>
      <div style={styles.div}>
        {initialQuestions &&
          answered.map((question) => {
            return <Poll key={question.id} question={question} value="View" />;
          })}
      </div>
    </div>
  );
};

const styles = {
  div: {
    display: "flex",
  },
  heading: {
    textAlign: "center",
  },
};

export default Dashboard;
