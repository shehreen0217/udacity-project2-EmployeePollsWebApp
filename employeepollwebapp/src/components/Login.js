import React, { useEffect, useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import Avatar from "@mui/material/Avatar";
import loginAvatar from "../avatars/loginavatar.jpg";
import "../index.css";
import Stack from "@mui/material/Stack";
import TextField from "@mui/material/TextField";
import Autocomplete from "@mui/material/Autocomplete";
import Button from "@mui/material/Button";
import { _getUsers } from "../_DATA";
import { Link } from "react-router-dom";
import { login } from "../Store/Actions/login";

const Login = () => {
  const users = ["sarahedo", "tylermcginnis", "mtsamis", "zoshikanlu"];
  const dispatch = useDispatch();
  const auth = useSelector((state) => state.auth.isAuthenticated);
  const [user, setUser] = useState("");
  const [password, setPassword] = useState("");

  const [reload, setReload] = useState(false);

  useEffect(() => {
    const loginHandler = async () => {
      console.log("username from component: ", user);
      dispatch(login(user, password));

      setReload(false);
    };
    loginHandler();
  }, [reload]);

  const clickHandler = () => {
    setReload(true);
  };

  return (
    !auth && (
      <div className="Login">
        <Stack spacing={2}>
          <Avatar
            className="avatar"
            alt="login avatar"
            src={loginAvatar}
            sx={{ width: 150, height: 150 }}
          />
          <h1 className="login-text">Login</h1>
          <Autocomplete
            inputValue={user}
            onInputChange={(e, currentUser) => {
              setUser(currentUser);
            }}
            disablePortal
            id="combo-box-demo"
            options={users}
            sx={{ width: 300 }}
            renderInput={(params) => <TextField {...params} label="User" />}
          />
          <TextField
            onChange={(e) => {
              setPassword(e.target.value);
            }}
            name="password"
            type="password"
            placeholder="password"
            label="Password"
          />
          <Link to="/home">
            {user && password && (
              <Button
                sx={{
                  mt: 1, // margin top
                }}
                onClick={clickHandler}
              >
                Log in
              </Button>
            )}
            {!user ||
              (!password && (
                <Button
                  disabled
                  sx={{
                    mt: 1, // margin top
                  }}
                  onClick={clickHandler}
                >
                  Log in
                </Button>
              ))}
          </Link>
        </Stack>
      </div>
    )
  );
};

export default Login;
