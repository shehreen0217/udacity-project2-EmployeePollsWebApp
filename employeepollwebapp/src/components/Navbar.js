import React from "react";
import { Link, NavLink } from "react-router-dom";
import classes from "../nav.module.css";
import { useDispatch, useSelector } from "react-redux";
import { authActions } from "../Store/reducers/login-slice";
import { userActions } from "../Store/reducers/user-slice";
import Avatar from "@mui/material/Avatar";
import { Stack } from "@mui/material";

const Navbar = () => {
  const isAuth = useSelector((state) => state.auth.isAuthenticated);
  const isUser = useSelector((state) => state.user.user);
  console.log(isUser);
  const dispatch = useDispatch();
  const logoutHandler = () => {
    dispatch(authActions.logout());
  };
  return (
    <header className={classes.header}>
      {isAuth && (
        <nav>
          <ul>
            <li>
              <NavLink activeClassName={classes.active} to="/home">
                home
              </NavLink>
            </li>
            <li>
              <NavLink activeClassName={classes.active} to="/leaderboard">
                Leaderboard
              </NavLink>
            </li>
            <li>
              <NavLink activeClassName={classes.active} to="/question">
                New
              </NavLink>
            </li>
            <span>
              <li style={styles.logout}>
                <NavLink
                  activeClassName={classes.active}
                  to="/"
                  onClick={logoutHandler}
                >
                  logout
                </NavLink>
              </li>
              <li>
                <Stack>
                  <Link to="/profile">
                    <Avatar
                      className="avatar"
                      alt="login avatar"
                      src={isUser[0].avatarURL}
                      sx={{ width: 50, height: 50 }}
                    />
                    {isUser[0].name}
                  </Link>
                </Stack>
              </li>
            </span>
          </ul>
        </nav>
      )}
    </header>
  );
};

const styles = {
  logout: {
    float: "right",
  },
  icon: {
    padding: "0px",
  },
};

export default Navbar;
