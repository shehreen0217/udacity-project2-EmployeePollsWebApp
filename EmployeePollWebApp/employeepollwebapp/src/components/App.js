import Login from "./Login";
import { useSelector } from "react-redux";
import Dashboard from "./Dashboard";
import Navbar from "./Navbar";
import { Route, Redirect } from "react-router-dom";
import PollDetails from "./PollDetails";
import LeaderBoard from "./LeaderBoard";
import CreatePoll from "./CreatePoll";

function App() {
  const auth = useSelector((state) => state.auth.isAuthenticated);
  return (
    <div>
      {!auth && (
        <Route to="/login">
          <Login />
        </Route>
      )}

      {auth && (
        <div>
          <Route path="/">
            <Navbar />
          </Route>
          <Route exact path="/add">
            <CreatePoll />
          </Route>
          <Route exact path="/">
            <Dashboard />
          </Route>
          <Route exact path="/question/:questionid">
            <PollDetails />
          </Route>
          <Route exact path="/leaderboard">
            <LeaderBoard />
          </Route>
        </div>
      )}
      {!auth && <Redirect to="/login" />}
    </div>
  );
}

export default App;
