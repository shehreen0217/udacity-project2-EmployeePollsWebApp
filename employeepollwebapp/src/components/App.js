import Login from "./Login";
import { useSelector } from "react-redux";
import Dashboard from "./Dashboard";
import Navbar from "./Navbar";
import { Route, Redirect } from "react-router-dom";
import PollDetails from "./PollDetails";
import LeaderBoard from "./LeaderBoard";
import CreatePoll from "./CreatePoll";

function App() {
  const auth = useSelector((state) => state.auth.isAuthenticated);
  return (
    <div>
      {!auth && <Login />}

      {auth && (
        <div>
          <Route path="/">
            <Navbar />
          </Route>
          <Route exact path="/add">
            <CreatePoll />
          </Route>
          <Route exact path="/home">
            <Dashboard />
          </Route>
          <Route exact path="/question/:questionid">
            <PollDetails />
          </Route>
          <Route exact path="/leaderboard">
            <LeaderBoard />
          </Route>
        </div>
      )}
      {!auth && <Redirect to="/" />}
    </div>
  );
}

export default App;
