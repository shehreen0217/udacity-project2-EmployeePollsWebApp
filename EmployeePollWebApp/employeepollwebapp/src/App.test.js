import {
  _saveQuestion,
  _saveQuestionAnswer,
  _getQuestions,
  _getUsers,
} from "./_DATA";
import LeaderBoard from "./components/LeaderBoard";
import App from "./components/App";
import { Provider } from "react-redux";
import store from "./Store/reducers";
import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router";
import Navbar from "./components/Navbar";
import Dashboard from "./components/Dashboard";
import Login from "./components/Login";

describe("testing async function SaveQuestion", () => {
  it("will return an object if correct values are passed", async () => {
    let question = {
      optionOneText: "work from home",
      optionTwoText: "work from office",
      author: "sarahedo",
    };
    let result = await _saveQuestion(question);
    expect(result.optionOne.text).toEqual(question.optionOneText);
    expect(result.optionTwo.text).toEqual(question.optionTwoText);
    expect(result.author).toEqual(question.author);
  });

  it("will reject the promise and throw error if incorrect values are passed", async () => {
    let question = {
      optionOneText: "work from home",
      optionTwoText: "work from office",
    };
    await expect(_saveQuestion(question)).rejects.toEqual(
      "Please provide optionOneText, optionTwoText, and author"
    );
  });
});

describe("testing async function SaveQuestionAnswer", () => {
  it("will resolve with true if correct values are passed", async () => {
    let authedUser = "sarahedo";
    let questionid = "vthrdm985a262al8qx3do";
    let answer = "optionOne";
    let result = await _saveQuestionAnswer({
      authedUser: authedUser,
      qid: questionid,
      answer: answer,
    });
    expect(result).toEqual(true);
  });

  it("will reject the promise and throw error if incorrect values are passed", async () => {
    let authedUser = "sarahedo";
    let questionid = "vthrdm262al8qx3do";

    await expect(
      _saveQuestionAnswer({
        authedUser: authedUser,
        qid: questionid,
      })
    ).rejects.toEqual("Please provide authedUser, qid, and answer");
  });
});

describe("testing _Users and _Questions", () => {
  it("will return all users from database", async () => {
    await expect(_getUsers()).not.toBeNull();
  });

  it("will return all questions from database", async () => {
    await expect(_getQuestions()).not.toBeNull();
  });
});

describe("testing snapshot", () => {
  it("App component", () => {
    let component = render(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });

  it("Leaderboard Component", () => {
    let component = render(
      <MemoryRouter>
        <Provider store={store}>
          <LeaderBoard />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });

  it("Navbar ", () => {
    let component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});

describe("testing DOM", () => {
  it("Login", () => {
    let component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Login />

          <Dashboard></Dashboard>
        </Provider>
      </MemoryRouter>
    );
    let userid = component.getByTestId("username").querySelector("input");
    fireEvent.change(userid, { target: { value: "tylermcginnis" } });
    let password = component.getByTestId("password").querySelector("input");
    fireEvent.change(password, { target: { value: "abc321" } });
    let submitButtom = component.getByRole("button");
    fireEvent.click(submitButtom);
  });
});
