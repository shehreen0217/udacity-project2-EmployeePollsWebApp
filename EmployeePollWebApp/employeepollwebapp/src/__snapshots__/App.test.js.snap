// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing snapshot App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="Login"
        >
          <div
            class="css-1p5q5e5-MuiStack-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular avatar css-cxgc6-MuiAvatar-root"
            >
              <img
                alt="login avatar"
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="loginavatar.jpg"
              />
            </div>
            <h1
              class="login-text"
            >
              Login
            </h1>
            <form>
              <div
                class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-viou3o-MuiAutocomplete-root"
                data-testid="username"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="combo-box-demo"
                    id="combo-box-demo-label"
                  >
                    User
                  </label>
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="off"
                      class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                      id="combo-box-demo"
                      role="combobox"
                      spellcheck="false"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="Open"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                        tabindex="-1"
                        title="Open"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="ArrowDropDownIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 10l5 5 5-5z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1ftyaf0"
                      >
                        <span>
                          User
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root css-9ahdq0-MuiFormControl-root-MuiTextField-root"
                data-testid="password"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r2:"
                  id=":r2:-label"
                >
                  Password
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r2:"
                    name="password"
                    placeholder="password"
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Password
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </form>
            <a
              href="/"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="Login"
      >
        <div
          class="css-1p5q5e5-MuiStack-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular avatar css-cxgc6-MuiAvatar-root"
          >
            <img
              alt="login avatar"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="loginavatar.jpg"
            />
          </div>
          <h1
            class="login-text"
          >
            Login
          </h1>
          <form>
            <div
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-viou3o-MuiAutocomplete-root"
              data-testid="username"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="combo-box-demo"
                  id="combo-box-demo-label"
                >
                  User
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="off"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="combo-box-demo"
                    role="combobox"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Open"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                      tabindex="-1"
                      title="Open"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        User
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root css-9ahdq0-MuiFormControl-root-MuiTextField-root"
              data-testid="password"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r2:"
                id=":r2:-label"
              >
                Password
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r2:"
                  name="password"
                  placeholder="password"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </form>
          <a
            href="/"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing snapshot Leaderboard Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
          >
            <table
              align="center"
              aria-label="customized table"
              class="MuiTable-root css-zwj2lt-MuiTable-root"
              id="leaderboard-table"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1enotpu-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1wvnaxz-MuiTableCell-root"
                    scope="col"
                  >
                    Users
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1nslmh4-MuiTableCell-root"
                    scope="col"
                  >
                    Answered
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1nslmh4-MuiTableCell-root"
                    scope="col"
                  >
                    Created
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              />
            </table>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
        >
          <table
            align="center"
            aria-label="customized table"
            class="MuiTable-root css-zwj2lt-MuiTable-root"
            id="leaderboard-table"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1enotpu-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1wvnaxz-MuiTableCell-root"
                  scope="col"
                >
                  Users
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1nslmh4-MuiTableCell-root"
                  scope="col"
                >
                  Answered
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1nslmh4-MuiTableCell-root"
                  scope="col"
                >
                  Created
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            />
          </table>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing snapshot Navbar  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
